File: Find All K-Distant Indices in an Array/Find All K-Distant Indices in an Array.py
# Solution for Find All K-Distant Indices in an Array in PY

class Solution:
    def findKDistantIndices(self, nums: List[int], key: int, k: int) -> List[int]:
        k_distant_indices = []
        n = len(nums)
        last_added = -1

        for j in range(n):
            if nums[j] == key:
                start = max(0, j - k)
                end = min(n - 1, j + k)

                if k_distant_indices and last_added >= start:
                    start = last_added + 1

                for i in range(start, end + 1):
                    k_distant_indices.append(i)
                    last_added = i

        return k_distant_indices

File: Find All K-Distant Indices in an Array/Find All K-Distant Indices in an Array.java
// Solution for Find All K-Distant Indices in an Array in JAVA

class Solution {
    public List<Integer> findKDistantIndices(int[] nums, int key, int k) {
        List<Integer> kDistantIndices = new ArrayList<>();
        int n = nums.length;
        int lastAdded = -1;

        for (int j = 0; j < n; j++) {
            if (nums[j] == key) {
                int start = Math.max(0, j - k);
                int end = Math.min(n - 1, j + k);

                // Skip duplicates
                if (!kDistantIndices.isEmpty() && lastAdded >= start) {
                    start = lastAdded + 1;
                }

                for (int i = start; i <= end; i++) {
                    kDistantIndices.add(i);
                    lastAdded = i;
                }
            }
        }

        return kDistantIndices;
    }
}
